src/storage/src/source/postgres.rs

```diff
@@ -645,37 +678,36 @@ impl PostgresTaskInfo {
         Ok(())
     }
 
-    /// Converts a Tuple received in the replication stream into a Row instance. The logical
-    /// replication protocol doesn't use the binary encoding for column values so contrary to the
-    /// initial snapshot here we need to parse the textual form of each column.
-    ///
-    /// The `old_tuple` argument can be used as a source of data to use when encountering unchanged
-    /// TOAST values.
-    fn row_from_tuple<'a, T>(rel_id: u32, tuple_data: T) -> Result<Row, anyhow::Error>
+    /// Packs a Tuple received in the replication stream into a Row packer.
+    fn datums_from_tuple<'a, T>(rel_id: u32, tuple_data: T) -> Result<Vec<Datum<'a>>, anyhow::Error>
     where
         T: IntoIterator<Item = &'a TupleData>,
     {
+        let mut datums = vec![];
```

Comment:
even better would be to use `DatumVec` and re-use the allocation. Actually that might fix our memory problems in the replication stream

Reply:
I had forgotten about `DatumVec`! Will switch to that